### PR TITLE
Add IssueCommentCommands

### PR DESCRIPTION
**Pull request by ludeeus**
This adds commands to GitHub issue comments.

To initialize it, 3 commands are added to aid non-member code owners in triage.

- Have the bot close the issue with `@home-assistant close`
  - This is limited to when there is only 1 integration labeled
- Have the bot change the title of the issue with `@home-assistant rename Awesome new title`
  - This is limited to when there is only 1 integration labeled
- Have the bot remove the integration label and assignees with `@home-assistant unassign`
  - This is limited to when there is only 1 integration labeled or the command is invoked with a domain argument `@home-assistant unassign domain`

The `CodeownersMention` handler will list out the commands that can be used based on the context.

Replaces https://github.com/home-assistant/service-hub/pull/122

### CODE REVIEW

File: services/bots/src/github-webhook/handlers/issue_comment_commands/commands.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+export const ISSUE_COMMENT_COMMANDS: { [command: string]: IssueCommentCommand } = {
+  close: {
+    description: 'Closes the issue.',
+    invokerType: 'code_owner',
+    handler: async (
+      context: WebhookContext<IssueCommentCreatedEvent>,
+      command: IssueCommentCommandContext,
+    ) => {
+      if (invokerIsCodeOwner(command)) {
```
Comment (balloob) [inline services/bots/src/github-webhook/handlers/issue_comment_commands/commands.ts:14]:
```suggestion
      if (!invokerIsCodeOwner(command)) {
```

File: services/bots/src/github-webhook/handlers/issue_comment_commands/commands.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  rename: {
+    description: 'Change the title of the issue.',
+    exampleAdditional: 'Awesome new title',
+    invokerType: 'code_owner',
+    requireAdditional: true,
+    handler: async (
+      context: WebhookContext<IssueCommentCreatedEvent>,
+      command: IssueCommentCommandContext,
+    ) => {
+      if (invokerIsCodeOwner(command)) {
```
Comment (balloob) [inline services/bots/src/github-webhook/handlers/issue_comment_commands/commands.ts:34]: Same
```suggestion
      if (!invokerIsCodeOwner(command)) {
```

File: services/bots/src/github-webhook/handlers/issue_comment_commands/commands.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        throw new Error('Only the issue creator can rename the issue.');
```
Comment (balloob) [inline services/bots/src/github-webhook/handlers/issue_comment_commands/commands.ts:35]: Should this be code owner ?
```suggestion
        throw new Error('Only the code owner can rename the issue.');
```

**Review by balloob — APPROVED** [review]
Nice!

File: services/bots/src/github-webhook/handlers/issue_comment_commands/commands.ts (hunk 2 above)
Comment (ludeeus) [reply services/bots/src/github-webhook/handlers/issue_comment_commands/commands.ts]: one more found while writing the tests for it <https://github.com/home-assistant/service-hub/pull/123/commits/0c2be776b1b43a49b35544a9348ca22b83cfc94c#diff-a75c2dcee499b4315a0bd7b2409e221cfadc4c0e5fdd09f370cf2b62550313fbR51>

**Pull request merged**